src/main/java/com/commercetools/project/sync/SyncerFactory.java

```diff
@@ -105,10 +106,13 @@ public static SyncerFactory of(
                               .sync(runnerNameOptionValue, isFullSync)
                               .toCompletableFuture(),
                           CustomerSyncer.of(sourceClient, targetClient, clock)
+                              .sync(runnerNameOptionValue, isFullSync)
+                              .toCompletableFuture(),
+                          ShoppingListSyncer.of(sourceClient, targetClient, clock)
```

Comment:
This looks wrong to me, Shopping List has a dependency on customer and types and products variants, so it's the wrong order, and this should be the latest one to run after the product sync is done.

Reply:
@ahmetoz, Okay I can correct this for now. But I suggest to have one Non regression test to know that there is a dependency on the order. Let me know how can this be handled?

Reply:
You have a line item in a shopping list, which is referencing a product variant, if the variant is not existed the sync will fail, to avoid it you need to sync the product first then the shopping lists, also not parallel but after all products are synced. So order matters. I am not sure what you meant by the non-regression test?

Reply:
Okay I understand the order now. Yes, I was talking about the test for this, to make sure the order is properly maintainted. So In future someone should not change the order again.

Reply:
You might add, but the concept is clear and documented in the readme, maybe try and skip the test, if it's hard to implement. But the order has to be adjusted.

Reply:
You should add a test for this. It's straightforward. Just try to sync a shopping list from a source project to a target project. This shopping list must contain a customer, a lineItem and a custom field. That's it 🙂 Let me know if you need some help.

Reply:
Integration test updated to test this scenario And changed the order as well. https://github.com/commercetools/commercetools-project-sync/pull/203/commits/0924e8c7a167c6ac223e871c9e9164e4c81b87ec